### PR TITLE
Database improvements

### PR DESCRIPTION
**Pull request by D4nte**
Few things on the DB:
- Quickcheck tests
- Store values in CBOR instead of JSON: The main reason is that it's a binary format because the DB stores byte arrays. It made less sense to use JSON than CBOR but there is little different considering the usecase.
- flush the db when storing a swap (arguably the most import patch of the PR)

### CODE REVIEW

File: src/network.rs [hunk 1]
```diff
@@ -632,3 +632,22 @@ mod transport {
...
     }
 }
+
+#[cfg(test)]
+mod arbitrary {
+    use super::*;
+    use libp2p::multihash;
+    use quickcheck::{Arbitrary, Gen};
+
+    impl Arbitrary for Taker {
```
Comment (thomaseizinger) [inline src/network.rs:642]: I can recommend looking into proptest as an alternative to quickcheck: https://docs.rs/proptest/0.10.0/proptest/

To nice thing is, it doesn't really on traits being implemented, you are just composing functions together.

Comment (D4nte) [reply src/network.rs:642]: proptest does not support async test, it's why I opted for quickcheck.

Comment (thomaseizinger) [reply src/network.rs:642]: ah bummer :/

**Pull request merged**